src/devices/cpu/drcbex86.cpp

```diff
@@ -5839,25 +5845,25 @@ void drcbe_x86::op_fload(Assembler &a, const instruction &inst)
 	// immediate index
 	if (indp.is_immediate())
 	{
-		a.mov(eax, MABS(basep.memory(4*indp.immediate())));                             // mov   eax,[basep + 4*indp]
+		a.mov(eax, MABS(basep.memory(inst.size()*indp.immediate())));
 		if (inst.size() == 8)
-			a.mov(edx, MABS(basep.memory(4 + 4*indp.immediate())));                     // mov   edx,[basep + 4*indp + 4]
+			a.mov(edx, MABS(basep.memory(4 + inst.size()*indp.immediate())));
 	}
 
 	// other index
 	else
 	{
 		Gp const indreg = indp.select_register(ecx);
 		emit_mov_r32_p32(a, indreg, indp);
-		a.mov(eax, ptr(u64(basep.memory(0)), indreg, 2));                               // mov   eax,[basep + 4*indp]
+		a.mov(eax, ptr(u64(basep.memory(0)), indreg, inst.size() == 8 ? 3 : 2));
 		if (inst.size() == 8)
-			a.mov(edx, ptr(u64(basep.memory(4)), indreg, 2));                           // mov   edx,[basep + 4*indp + 4]
+			a.mov(edx, ptr(u64(basep.memory(4)), indreg, inst.size() == 8 ? 3 : 2));
```

Comment:
Can you please put explicit parentheses around non-trivial arguments of ternary conditional operators?  They can be difficult to read otherwise.  (Like `(inst.size() == 8) ? 3 : 2`.)